samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -64,13 +64,11 @@
   private volatile String ephemeralPath = null;
   private final ZkKeyBuilder keyBuilder;
   private final int connectionTimeoutMs;
-  private final String processorId;
 
-  public ZkUtils(String processorId, ZkKeyBuilder zkKeyBuilder, ZkClient zkClient, int connectionTimeoutMs) {
```

Comment:
Very minor: Slightly prefer `utils` classes as a collection of `static` methods instead of classes that contain state. Can rename this to `ZkClientWrapper` or something similar.